src/liballoc/rc.rs

```diff
@@ -312,6 +312,43 @@ impl<T> Rc<T> {
         }
     }
 
+    /// Construct a new Rc with uninitialized contents.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(new_uninit)]
+    /// #![feature(get_mut_unchecked)]
+    ///
+    /// use std::rc::Rc;
+    ///
+    /// let mut five = Rc::<u32>::new_uninit();
+    ///
+    /// let five = unsafe {
+    ///     // Deferred initialization:
+    ///     Rc::get_mut_unchecked(&mut five).as_mut_ptr().write(5);
+    ///
+    ///     five.assume_init()
+    /// };
+    ///
+    /// assert_eq!(*five, 5)
+    /// ```
+    #[unstable(feature = "new_uninit", issue = "0")]
+    pub fn new_uninit() -> Rc<mem::MaybeUninit<T>> {
+        let layout = Layout::new::<RcBox<mem::MaybeUninit<T>>>();
+        unsafe {
+            let mut ptr = Global.alloc(layout)
+                .unwrap_or_else(|_| handle_alloc_error(layout))
+                .cast::<RcBox<mem::MaybeUninit<T>>>();
```

Comment:
Couldn't this use `Box::new_uninit`?

Reply:
It could, but I’m generally not in favor of using `Box` as an allocator when we don’t actually want a box. To reduce the `handle_alloc_error` boilerplate I’d prefer we had dedicated infallible APIs.

Reply:
`Rc::new` goes through `box` though, seems reasonable to have both code paths be parallel.